packages/drivers/odsp-driver/src/zipItDataRepresentationUtils.ts

```diff
@@ -0,0 +1,416 @@
+    8: 9,
+};
+
+/**
+ * This contains mapping of number of bytes representing the corresponding string length to Marker Codes.
+*/
+const utf8StringBytesToCodeMap = {
+    0: 13,
+    1: 14,
+    2: 15,
+    4: 16,
+};
+
+/**
+ * This contains mapping of number of bytes representing the corresponding length in which actual data(base64 string)
+ * will be stored to Marker Codes.
+*/
+const binaryBytesToCodeMap = {
+    0: 32,
+    1: 33,
+    2: 34,
+    4: 35,
+    8: 16,
+};
+
+/**
+ * Calculate how many bytes are required to encode an integer. This is always multiple of 2.
+ * So if 6 bytes are required to store an integer than it will return 8.
+ * @param num - number to encode.
+ */
+function calcLength(numArg: number) {
+    let num = numArg;
+    let lengthLen = 0;
+    while (num > 0) {
+        num = Math.floor(num / 256);
+        lengthLen++;
+    }
+    let res = 0;
+    let index = 0;
+    while (res < lengthLen) {
+        res = Math.pow(2, index);
+        index++;
+    }
+    return res;
+}
+
+export function getAndValidateNodeProps<T>(node: NodeCore, props: string[]) {
+    const propSet = new Set(props);
+    const res: Record<string, NodeTypes> = {};
+    for (const [key, value] of node.iteratePairs()) {
+        assert(key instanceof BlobCore, "Prop name should be a blob");
+        const keyStr = key.toString();
+        assert(propSet.has(keyStr), "Property should exist");
+        propSet.delete(keyStr);
+        res[keyStr] = value;
+    }
+    assert(propSet.size === 0, "All properties should exist");
+    return res as unknown as T;
+}
+
+export function iteratePairs<T>(it: IterableIterator<T>) {
+    const res: IterableIterator<[T, T]> = {
+        next: () => {
+            const a = it.next();
+            if (a.done) {
+                return { value: undefined, done: true };
+            }
+            const b = it.next();
+            assert(b.done !== true, "Should be a pair");
+            return { value: [a.value, b.value], done: b.done };
+        },
+        [Symbol.iterator]: () => { return res; },
+    };
+    return res;
+}
+
+/**
+ * Helper function that returns iterator from an object
+ * @param obj - object that supports iteration
+ */
+export function iterate<T>(obj: {[Symbol.iterator]: () => IterableIterator<T>}) {
+    return obj[Symbol.iterator]();
+}
+
+/**
+ * Base class to represent binary blob element.
+ * Binary blob is one of three types supported as a leaf node of a tree.
+ * Note: concrete implementations (derived classes) are not exposed from this module
+ */
+export abstract class BlobCore {
+    public abstract get buffer(): Uint8Array;
+    /**
+     * Represents a blob.
+     * @param useUtf8Code - Represents if the utf8 string marker code should be used when representing.
+     */
+    constructor(private readonly useUtf8Code: boolean = false) {}
+
+    public toString() {
+        return Uint8ArrayToString(this.buffer, "utf-8");
+    }
+
+    public write(buffer: WriteBuffer) {
```

Comment:
I think I asked that question earlier: is there way to refactor code such that all write/serialize code is in separate file, for tree shaking to remove it?

Reply:
Refactored the code to separate the concerns.